src/tail/mod.rs

```diff
@@ -19,12 +31,29 @@ impl Tail {
         let mut runtime = TokioRuntime::new()?;
 
         runtime.block_on(async {
+            // Create three [one-shot](https://docs.rs/tokio/0.2.16/tokio/sync#oneshot-channel)
+            // channels for handling ctrl-c. Each channel has two parts:
+            // tx: Transmitter
+            // rx: Receiver
+            let (log_tx, log_rx) = oneshot::channel();
+            let (session_tx, session_rx) = oneshot::channel();
+            let (tunnel_tx, tunnel_rx) = oneshot::channel();
+
+            // Pass the three transmitters to a newly spawned sigint handler
+            let txs = vec![log_tx, tunnel_tx, session_tx];
+            let listener = tokio::spawn(listen_for_sigint(txs));
+
+            // Spin up a local http server to receive logs
+            let log_server = tokio::spawn(LogServer::new(log_rx).run());
+
+            // Spin up a new cloudflared tunnel to connect trace worker to local server
             let tunnel_process = Tunnel::new()?;
-            let res = tokio::try_join!(
-                Host::run(),
-                tunnel_process.run(),
-                Session::run(&target, &user)
-            );
+            let tunnel = tokio::spawn(tunnel_process.run(tunnel_rx));
+
+            // Register the tail with the Workers API and send periodic heartbeats
+            let session = tokio::spawn(Session::run(target, user, session_rx));
```

Comment:
```suggestion
            let keep_alive = tokio::spawn(Session::run(target, user, session_rx));
```

maybe keep_alive isn't the right name here but `session` feels... vague?

Reply:
idk what else to call it. it's the tail session with the api.